spec/lib/elasticsearch_client_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+require 'vcr_helper'
+
+require 'rex_releases'
+
+RSpec.describe ElasticsearchClient, vcr: VCR_OPTS do
+
+  before {
+    # Even in VCR playback, the code freaks out if these aren't set (in
+    # playback, values don't matter)
+    ENV['AWS_ACCESS_KEY_ID'] ||= 'foo'
+    ENV['AWS_SECRET_ACCESS_KEY'] ||= 'bar'
+  }
+
+  let(:fake_es_domain_name) { "spec-esdomain-#{SecureRandom.hex(7)}" }
+
+  it 'can access a restricted AWS ES domain using signed requests' do
+    TempAwsEnv.make do |env|
+      domain_status = env.create_elasticsearch_domain(name: fake_es_domain_name)
+
+      signing_client = ElasticsearchClient.new(
+        url: "https://#{domain_status.endpoint}",
+        sign_aws_requests: true
+      )
+
+      resp = signing_client.search q: "test"
+      expect(resp).to have_key("took")
+
+      non_signing_client = ElasticsearchClient.new(
+        url: "https://#{domain_status.endpoint}",
+        sign_aws_requests: false
+      )
+
+      expect {
+        non_signing_client.search q: "test"
+      }.to raise_error(Elasticsearch::Transport::Transport::Errors::Forbidden)
+    end
+  end
+
```

Comment:
@jpslav  do you care one way or the other about blank lines before and after class declarations/ends?  i see this pattern in your code.  however, i dont see this pattern in the ruby coding standards https://github.com/rubocop-hq/ruby-style-guide#indent-public-private-protected (it has suggestions on blank lines, but not just after class declarations/ends.  Asking if you care and if there's a pattern you want us to follow.  My 2 cents is to follow the ruby rubocop standards on this (which is no blank lines in these specific spots).

Reply:
My preference is to not enforce a rule here.  I probably do it both ways and I think I choose in the moment based on what looks clearer to me.  Here I wanted the before block to have a little more visual separation.  In the later example with the monkey patch, it was very short and I went with a compact look.